src/controllers/index.controller.js

```diff
@@ -112,13 +114,57 @@ async function start(req, res) {
                 + `and ${multisigData.numSignatures} numSignatures`);
   }
 
+  // Validate address scanning policy
+  let scanPolicyData = null;
+  if ('scanPolicy' in req.body) {
+    const policy = req.body.scanPolicy;
+    switch (policy) {
+      case 'index-limit':
+        /**
+         * The policy configuration is composed by:
+         * - policyStartIndex
+         *   - optional, defaults to 0
+         * - policyEndIndex
+         *   - optional, defaults to policyStartIndex
+         *
+         * If no configuration is passed, only the first address will be loaded.
+         *
+         * Obs: The route does not use validation or parser, so we need to parse the integers
+         */
+
+        // parseInt returns NaN (falsy) for null, undefined or malformed number strings
+        scanPolicyData = {
+          policy,
+          startIndex: parseInt(req.body.policyStartIndex, 10) || 0,
+        };
+        scanPolicyData.endIndex = parseInt(req.body.policyEndIndex, 10)
+          || scanPolicyData.startIndex;
+        break;
+      case 'gap-limit':
+        // The gapLimit is optional and will default to 20
+        scanPolicyData = {
+          policy,
+          gapLimit: parseInt(req.body.gapLimit, 10) || GAP_LIMIT,
+        };
+        break;
+      default:
+        // address scanning policy requested is not supported
+        res.send({
+          success: false,
+          message: `Address scanning policy ${policy} is not supported.`,
+        });
+        return;
+    }
+  }
+
   let walletConfig;
   if ('xpubkey' in req.body) {
     try {
       // starting a readonly wallet
       walletConfig = getReadonlyWalletConfig({
         xpub: req.body.xpubkey,
         multisigData,
+        scanPolicy: scanPolicyData,
```

Comment:
[⭕ verify] The `scanPolicyData` should be validated inside the `getReadlongWalletConfig` because it can be evaluated to non integer value after `parseInt`.

Reply:
It it is not an integer, it would return `NaN`, which is a falsy value and the or cases (`||`) would prevent non integers.

Although I still believe we should improve validation on this route, which should be done with `express-validator` a dependency already used on other routes.
The validator would check for int values and return an error message to the user if invalid data was received.
This is outside the scope of this PR, but I think we should create validators for all routes that receive user input.

Reply:
Oh! You right. I miss the `||` operator.